File: typescript/6kyu/Your_order_please.ts
export function order(words: string): string {
  const wordsObj: { [key: string]: number } = {};

  words.split(" ").forEach((word) =>
    word.split("").forEach((el) => {
      if (!isNaN(parseInt(el))) wordsObj[word] = parseInt(el);
    })
  );
  const sortedWords = Object.fromEntries(
    Object.entries(wordsObj).sort(([, a], [, b]) => a - b)
  );

  return Object.keys(sortedWords).join(" ");
}

File: python/6kyu/Your_order_please.py

def order(sentence):
    sentence_dict = {}

    for word in sentence.split(" "):
        for el in list(word):
            if el.isdigit():
                sentence_dict[word] = int(el)
                
    sorted_sentence_dict = dict(sorted(sentence_dict.items(), key=lambda item: item[1]))
    
    return " ".join(list(sorted_sentence_dict.keys()))